core/src/main/java/com/novoda/noplayer/internal/exoplayer/NoPlayerAdsLoader.java

```diff
@@ -179,39 +182,54 @@ public void onTimelineChanged(Timeline timeline, @Nullable Object manifest, int
             return;
         }
 
-        // handle only pre-roll start on timeline change
-        if (reason == Player.TIMELINE_CHANGE_REASON_PREPARED) {
-            handleAdvertStart();
-        }
+        updateState();
     }
 
-    private void handleAdvertStart() {
-        if (advertHasNotStarted()) {
-            adGroupIndex = player.getCurrentAdGroupIndex();
-            adIndexInGroup = player.getCurrentAdIndexInAdGroup();
+    private void updateState() {
```

Comment:
In this solution the `AdvertIndexInAdvertGroup` and the `isPlaying` determines how everything runs. In the event that `multiple calls` are made with the same index the new value will not be different and therefore will not retrigger events. If the index is different then we determine whether we were playing an advert in the last call and potentially finish the advert / advert break.